clang/lib/Sema/AnalysisBasedWarnings.cpp

```diff
@@ -696,47 +665,41 @@ static void CheckFallThroughForBody(Sema &S, const Decl *D, const Stmt *Body,
   if (CD.checkDiagnostics(Diags, ReturnsVoid, HasNoReturn))
       return;
   SourceLocation LBrace = Body->getBeginLoc(), RBrace = Body->getEndLoc();
-  auto EmitDiag = [&](SourceLocation Loc, unsigned DiagID) {
-    if (IsCoroutine)
-      S.Diag(Loc, DiagID) << FSI->CoroutinePromise->getType();
-    else
-      S.Diag(Loc, DiagID);
-  };
```

Comment:
Er, why are we getting rid of this? I think we *do* still want to print the coroutine promise type if the function is a coroutine.

Reply:
Actually, we weren’t even doing anything with the promise type were we? Oh well, just get rid of it then (i.e. leave your changes here as-is); I don’t think we care too much about that.

Reply:
It is not used anymore then there is no reason to keep it.